TypeError: Cannot read property 'nullable' of undefined

I'm running the npm as a pretest step in my tests:
"pretest": "openapi-typescript https://dev-e.compayname.com/swagger/Private-v1.0/swagger.json --output ./.temp/types.ts"

in the console, I'm getting:
```
✨ openapi-typescript 3.1.2
🔭 Loading spec from https://dev-e.companyname.com/swagger/Private-v1.0/swagger.json…
TypeError: Cannot read property 'nullable' of undefined
    at Object.transformSchemaObj (C:\compayname_autotests\node_modules\openapi-typescript\dist\cjs\transform\schema.js:28:14)      
    at C:\compayname_autotests\node_modules\openapi-typescript\dist\cjs\transform\operation.js:40:49
    at Array.forEach (<anonymous>)
    at transformRequestBodyObj (C:\compayname_autotests\node_modules\openapi-typescript\dist\cjs\transform\operation.js:39:33)     
    at Object.transformOperationObj (C:\compayname_autotests\node_modules\openapi-typescript\dist\cjs\transform\operation.js:27:28)
    at C:\compayname_autotests\node_modules\openapi-typescript\dist\cjs\transform\paths.js:28:64
    at Array.forEach (<anonymous>)
    at C:\compayname_autotests\node_modules\openapi-typescript\dist\cjs\transform\paths.js:17:79
    at Array.forEach (<anonymous>)
    at Object.transformPathsObj (C:\compayname_autotests\node_modules\openapi-typescript\dist\cjs\transform\paths.js:9:27)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! compayname_ui_api@1.0.0 pretest: `openapi-typescript https://dev-e.compayname.com/swagger/Private-v1.0/swagger.json --output ./.temp/types.ts`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the compayname_ui_api@1.0.0 pretest script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Name\AppData\Roaming\npm-cache\_logs\2021-03-23T01_46_08_311Z-debug.log
```


In the debug log I have:
```
22 verbose stack Error: command failed
22 verbose stack     at ChildProcess.<anonymous> (C:\Users\Name\AppData\Roaming\npm\node_modules\npm\node_modules\@npmcli\promise-spawn\index.js:64:27)
22 verbose stack     at ChildProcess.emit (events.js:315:20)
22 verbose stack     at maybeClose (internal/child_process.js:1048:16)
22 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)
23 verbose pkgid companyname_ui_api@1.0.0

```

Swagger.json:

```
{
  "openapi": "3.0.1",
  "info": {
    "title": "ComanyName Private API",
    "description": "ComanyName Private API",
    "termsOfService": "https://www.companyname.com/terms",
    "contact": {
      "name": "ComanyName" ,
      "url": "https://www.companyname.com/enterprise",
    },
    "license": {
      "name": "Use under license",
      "url": "https://www.companyname.com/terms"
    },
    "version": "v1.0"
```

Accessing open swagger API with the command:
```
npx openapi-typescript https://enterprise.skyhive.io/swagger/v1.0/swagger.json --output test-prod-api.ts
```
 lead to the same issue - TypeError: Cannot read property 'nullable' of undefined

1 info using npm@7.6.3
2 info using node@v14.16.0


Could you say, what can be wrong? How to fix it?

--- a/src/transform/responses.ts
+++ b/src/transform/responses.ts
@@ -47,10 +47,11 @@ export function transformResponsesObj(responsesObj: Record<string, any>, options
       // V3
       output += `    ${readonly}content: {\n`; // open content
       Object.entries(response.content).forEach(([contentType, contentResponse]) => {
-        output += `      ${readonly}"${contentType}": ${transformSchemaObj(
-          (contentResponse as any).schema,
-          options
-        )};\n`;
+        const responseType =
+          contentResponse && (contentResponse as any).schema
+            ? transformSchemaObj((contentResponse as any).schema, options)
+            : "unknown";
+        output += `      ${readonly}"${contentType}": ${responseType};\n`;
       });
       output += `    }\n`; //close content
     } else if (response.schema) {

--- a/src/utils.ts
+++ b/src/utils.ts
@@ -63,6 +63,7 @@ export function nodeType(obj: any): SchemaObjectType | undefined {
 export function swaggerVersion(definition: OpenAPI2 | OpenAPI3): 2 | 3 {
   // OpenAPI 3
   if ("openapi" in definition) {
+    // OpenAPI version requires semver, therefore will always be string
     if (parseInt(definition.openapi, 10) === 3) {
       return 3;
     }
@@ -70,6 +71,10 @@ export function swaggerVersion(definition: OpenAPI2 | OpenAPI3): 2 | 3 {
 
   // OpenAPI 2
   if ("swagger" in definition) {
+    // note: swagger 2.0 may be parsed as a number
+    if (typeof definition.swagger === "number" && Math.round(definition.swagger as number) === 2) {
+      return 2;
+    }
     if (parseInt(definition.swagger, 10) === 2) {
       return 2;
     }

--- a/tests/paths.test.ts
+++ b/tests/paths.test.ts
@@ -155,11 +155,17 @@ describe("transformPathsObj", () => {
 }\n`);
   });
 
-  it("empty responses (#333)", () => {
+  it("empty responses (#333, #536)", () => {
     const emptyResponsesSchema = {
       "/no-content": {
         get: {
           responses: {
+            200: {
+              description: "OK",
+              content: {
+                "application/json": {},
+              },
+            },
             204: {
               description: "Empty response",
             },
@@ -190,6 +196,12 @@ describe("transformPathsObj", () => {
   "/no-content": {
     get: {
       responses: {
+        /** OK */
+        200: {
+          content: {
+            "application/json": unknown;
+          };
+        };
         /** Empty response */
         204: never;
       };
@@ -217,6 +229,12 @@ describe("transformPathsObj", () => {
   readonly "/no-content": {
     readonly get: {
       readonly responses: {
+        /** OK */
+        readonly 200: {
+          readonly content: {
+            readonly "application/json": unknown;
+          };
+        };
         /** Empty response */
         readonly 204: never;
       };

